src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -781,6 +855,16 @@ std::string postQueryContext
     }
   }
 
+  if ((ciP->apiVersion == V2) && (ciP->uriParamOptions["count"]))
+  {
+    char cV[32];
+
+    snprintf(cV, sizeof(cV), "%llu", *countP);
+
+    ciP->httpHeader.push_back(HTTP_FIWARE_TOTAL_COUNT);
+    ciP->httpHeaderValue.push_back(cV);
+  }
```

Comment:
It seems this block has been moved, from old L526 to this point. Could you elaborate on which that movement is needed?

Reply:
Yes, we have moved the block, because at old location i.e. at L#526, `ciP->httpHeader` vector were holding the **count of only CB entities**.
Now in new location i.e. at L#858, `ciP->httpHeader` vector is holding **count of all the entities including CB+CP**. Because Total count of all the entities (CP+CB) is available only after queryForward() function call i.e. at L#785.